C#/TollFeeCalculator/TollCalculator.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using TollFeeCalculator.Vehicles;
+
+namespace TollFeeCalculator
+{
+    public class TollCalculator
+    {
+        /// <summary>
+        /// Calculates the total fee for a vehicles passages
+        /// </summary>
+        /// <param name="vehicle">The vehicle</param>
+        /// <param name="passages">Timestamps the vehicles passed through the tolls.</param>
+        /// <returns>Total fee</returns>
+        public int GetTollFee(IVehicle vehicle, DateTime[] passages)
+        {
+            return passages
+                .GroupBy(x => x.Date)
+                .Sum(x => GetADaysTotalTollFee(vehicle, x.ToArray()));
+        }
+
+        /// <summary>
+        /// Calculate the total toll fee for one day
+        /// </summary>
+        /// <param name="vehicle">The vehicle</param>
+        /// <param name="passages">Timestamps the vehicles passed through the tolls.</param>
+        /// <returns>Total fee</returns>
+        /// <exception cref="ArgumentException">Throws argument exception if passages timestamps are on different days.</exception>
+        public int GetADaysTotalTollFee(IVehicle vehicle, DateTime[] passages)
+        {
+            if (passages.Length == 0)
+            {
+                return 0;
+            }
+            
+            if (passages.Select(x => x.Date).Distinct().Count() > 1)

```

Comment:
To be honest, I'm not a huge fan of throwing an exception here. I assumed this is a library that other developers are going to use. In that case, I think it's better for them to fail fast if they pass data that is incorrect.